libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -725,14 +725,17 @@ void NavEKF3_core::UpdateFilter(bool predict)
         statesInitialised = false;
         InitialiseFilterBootstrap();
     }
+
+    // check for update of earth field
+    checkUpdateEarthField();
```

Comment:
Could move this to bottom of if (runUpdates) { ... }

Reply:
fixed